packages/autorest.python/autorest/m4reformatter/__init__.py

```diff
@@ -430,8 +430,13 @@ def legacy(self) -> bool:
 
     @property
     def only_path_and_body_parameters_positional(self) -> bool:
-        return self.version_tolerant or bool(
-            self._autorestapi.get_boolean_value("only-path-and-body-params-positional")
+        return self.version_tolerant or (
+            bool(
+                self._autorestapi.get_boolean_value(
+                    "only-path-and-body-params-positional"
+                )
+            )
+            and not bool(self._autorestapi.get_boolean_value("azure-arm"))
```

Comment:
we shall keep `self.only_path_and_body_parameters_positional` with `False` otherwise credential order will be changed:
https://github.com/Azure/autorest.python/blob/96bdb21fea3fa46bbf7d62057939168c011f12fd/packages/autorest.python/autorest/m4reformatter/__init__.py#L1136-L1140
nit: in python `False or True` = `True`
![image](https://user-images.githubusercontent.com/70930885/227404424-577677bc-bb56-4088-8e02-ecf35cd00d76.png)